uaclient/timer/update_contract_info.py

```diff
@@ -28,3 +28,37 @@ def update_contract_info(cfg: UAConfig) -> bool:
             )
             return False
     return True
+
+
+def validate_release_series(cfg: UAConfig, show_message=False):
+    machine_token_file = machine_token.get_machine_token_file(cfg)
+    current_series = system.get_release_info().series
+    if current_series:
+        support_entitlement = (
+            machine_token_file.entitlements()
+            .get("support", {})
+            .get("entitlement")
+        )
+        if support_entitlement:
+            allowed_series = support_entitlement.get("affordances", {}).get(
+                "onlySeries"
+            )
+            if allowed_series and allowed_series != "none":
+                if allowed_series != current_series:
+                    machine_token_file.delete()
```

Comment:
This should probably call our `detach` api function

Reply:
Updated @orndorffgrant